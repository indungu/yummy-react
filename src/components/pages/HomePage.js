import React from 'react';
import { Redirect } from 'react-router-dom';
import NavBar from '../common/NavBar';
import Footer from '../common/Footer';

const HomePage = () => {
  const token = window.localStorage.getItem('token');
  if (token) {
    return <Redirect to="/dashboard" />;
  }
  return (
    <div className="App">
      <NavBar />
      <div className="jumbotron">
        <div className="container">
          <div className="row">
            <div className="col-md-6">
              <div className="well">
                <h1 className="App-title">Welcome foodie,</h1>
                <p className="lead">Yummy Recipes is a platform for you to keep track of your awesome recipes and share with others if you so wish.
                </p>
              </div>
            </div>
            <div className="col-md-3">
              <div className="well">
                <h3>New foodie?</h3>
                <a href="/signup" className="btn btn-outline-primary btn-block">Sign Up</a>
              </div>
            </div>
            <div className="col-md-3">
              <div className="well">
                <h3>Returning foodie?</h3>
                <a href="/login" className="btn btn-outline-success btn-block">Login</a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <Footer />
    </div>
  );
};

export default HomePage;
