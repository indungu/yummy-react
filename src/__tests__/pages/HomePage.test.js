import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import HomePage from '../../components/pages/HomePage';

describe('HomePage', () => {
  const wrapper = shallow(<HomePage />);
  it('Renders properly ', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
