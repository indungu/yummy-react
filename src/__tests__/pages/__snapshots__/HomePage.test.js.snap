// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Renders properly  1`] = `
<div
  className="App"
>
  <NavBar />
  <div
    className="jumbotron"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <div
            className="well"
          >
            <h1
              className="App-title"
            >
              Welcome foodie,
            </h1>
            <p
              className="lead"
            >
              Yummy Recipes is a platform for you to keep track of your awesome recipes and share with others if you so wish.
            </p>
          </div>
        </div>
        <div
          className="col-md-3"
        >
          <div
            className="well"
          >
            <h3>
              New foodie?
            </h3>
            <a
              className="btn btn-outline-primary btn-block"
              href="/signup"
            >
              Sign Up
            </a>
          </div>
        </div>
        <div
          className="col-md-3"
        >
          <div
            className="well"
          >
            <h3>
              Returning foodie?
            </h3>
            <a
              className="btn btn-outline-success btn-block"
              href="/login"
            >
              Login
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;
